sunpy/timeseries/timeseriesbase.py

```diff
@@ -530,19 +530,32 @@ def truncate(self, a, b=None, int=None):
         `~sunpy.timeseries.TimeSeries`
             A new `~sunpy.timeseries.TimeSeries` with only the selected times.
         """
+        is_int = False
         # Evaluate inputs
         # If given strings, then use to create a sunpy.time.timerange.TimeRange
         # for the SunPy text date parser.
         if isinstance(a, str) and isinstance(b, str):
-            a = TimeRange(a, b)
-        if isinstance(a, TimeRange):
+            time_range = TimeRange(a, b)
+            start = time_range.start.datetime
+            end = time_range.end.datetime
+        elif isinstance(a, TimeRange):
             # If we have a TimeRange, extract the values
             start = a.start.datetime
             end = a.end.datetime
         else:
             # Otherwise we already have the values
             start = a
             end = b
+            is_int = True
+
+        min_time, max_time = self._data.index.min(), self._data.index.max()
+        # Check if the timerange overlaps with the data timerange
+        if not is_int:
+            if min_time >= start and max_time <= end:
+                pass
+            elif not ((min_time <= start <= max_time) or (min_time <= end <= max_time)):
+                message = "Provided timerange is not within the bounds of the timeseries"
+                raise ValueError(message)
```

Comment:
Is there a need for the if/elif?

Reply:
yes
if checks ` (min_time, max_time)` is subset of `(start ,end)` or not , if yes then `timeseries.truncate` will not be truncated as it encloses whole time range
 
 elif checks - either start and end lie inside (min_time, max_time) or (start ,end) lie inside (min_time, max_time)
 it make sure if first statement executes then it shouldn't